Modules/database_manager.py

```diff
@@ -0,0 +1,432 @@
+                      "PRIMARY KEY (name, lang) "
+                      #  "UNIQUE (name, lang)"
+                      ");"
+                      )
+
+        self._execute("CREATE TABLE IF NOT EXISTS fact_timestamps ("
+                      "name VARCHAR NOT NULL,"
+                      "lang VARCHAR NOT NULL,"
+                      "last_modified TIMESTAMP WITH TIME ZONE NOT NULL,"
+                      "PRIMARY KEY (name, lang)"
+                      ");"
+                      )
+
+    def __init__(self):
+        """
+        Connects to the DB, sets up the connection, retrieves the cursor.
+        Creates the default tables should they not exist.
+        """
+        # if we are not supposed to connect on start-up, we just sit tight and do nothing
+        if not self.enabled:
+            return
+        # we init the connection, as we are supposed to do
+        try:
+            self.init_connection()
+        except pyodbc.Error:
+            # something went terribly wrong and excrements hit the ventilation device
+            # using internal here to skip disconnection part as it might (and probably would) call
+            # stuff on a NoneType
+            self._enabled = False
+            return
+
+    async def select_rows(self, table_name: str, connector: str, condition: dict = None,
+                          skip_double_dash_test: bool = False) -> list or None:
+        """
+
+        Args:
+            table_name: name of the table to select from
+            connector: Connector used to connect conditions. Must be supported by the DB
+            condition: conditions, connected by "equals"
+            skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                    implement your OWN CHECK!
+
+        Returns: a list containing the rows returned
+
+        """
+        # if we don't have that table, somebody didn't do their homework, so we just raise them.
+        if await self.has_table(table_name):
+            if not condition:
+                condition = {}
+            # we need the conditions written out as a string
+            cond_str = f" {connector} ".join(f"{key} = '{value}'"
+                                             for key, value in condition.items())
+            # this is just a crude check trying to prevent SQL-Injection by disallowing comments
+            if ("--" in cond_str) and not skip_double_dash_test:
+                raise ValueError("Suspicion of SQL-Injection. Statement: SELECT * FROM {table_name}"
+                                 f" WHERE {cond_str}. Aborting")
+            # throw out the result of the query
+            return self._execute(f"SELECT * FROM {table_name} WHERE {cond_str};").fetchall()
+        else:
+            # let's raise them to be responsible adults
+            raise ValueError(f"Table {table_name} does not exists!")
+
+    async def has_table(self, name: str) -> bool:
+        """

```

Comment:
As I recall, docstrings done in the ABC need not be re-defined in the implementation.

Reply:
they do not need to be redefined, but it can't hurt to have them there again, as it's the function/method/procedure you will be pointed at by your IDE / the first to look at if you want to understand it a bit better, need docs. also I''m pretty sure it missing might break ctrl+q.

Reply:
Doesn't break ctrl+q in pycharm at any rate, its smart and copies the docstring from the ABC.

but yeh, i suppose it doesn't hurt until the docstrings don't add up. right now they are duplicated, but what of the future?